base/ltxref.dtx

```diff
@@ -343,16 +393,27 @@
 %    to something like |\foo{4.d}|.  Change made 20 Jul 88.
 %
 %    \begin{macrocode}
+%</2ekernel>
+%<*2ekernel|latexrelease>
+%<latexrelease>\IncludeInRelease{2023/06/01}%
+%<latexrelease>                 {\label}{store five label arguments}%
+\providecommand\@currentlabelname{}
+\providecommand\@currentHref{}
+\providecommand\@kernel@reserved@label@data{}
+\NewHookWithArguments{label}{1}
 \def\label#1{\@bsphack
+  %\begingroup
+  \UseHookWithArguments{label}{1}{#1}%
   \protected@write\@auxout{}%
-         {\string\newlabel{#1}{{\@currentlabel}{\thepage}}}%
+         {\string\newlabel{#1}{{\@currentlabel}{\thepage}%
+          {\@currentlabelname}{\@currentHref}{\@kernel@reserved@label@data}}}%
+  %\endgroup        
   \@esphack}
```

Comment:
ups, good catch. I will add it (it is no longer strictly needed to protect the \@bshack value but we decided to keep it).

Reply:
Also, it would be good to change the wording in ltnews37 as follows, to be more precise:

. . . thus its code can be made local, so as to affect only the write operation.